### ISSUE TITLE
Sharding metrics

### ISSUE THREAD

**Issue by penguinlav** [avito-tech/bioyino#71]
Hi! Our bioyino cluster consumes a lot of resources from each host. Probably, the cluster node will run out of resources soon. Are there any plans to shard the collected metrics?

**Comment by Albibek** [comment]
Hi. Are you using agent-based approach so far? You probably could shard your metrics per-agent first.

**Comment by penguinlav** [comment]
Yes, we already use this approach with local aggregate nodes. And we are close to the limit of resources.
We can reduce period of time between sending snapshots (2 sec). As far I can see, it will help reduce cpu consumption for snapshot merging by reducing the number of snapshots in the `interval` (send metrics to carbon).
But what if there is a sharding feature? Unlimited ability to increase the number of metrics collected by bioyino :)

**Comment by Albibek** [comment]
To be clear, snapshots interval is regulated by `snapshot-interval`, not by `interval`, but you are right, it may reduce some load because of less TCP connection overheads and a lower number of total TCP connections.

Sharding is not so easy when it comes to failing nodes. Let's say you have a 3-node cluster and 10 agents. How would you distribute data among them and how you expect to configure it? What behavior do you expect when 1 cluster node or 2 cluster nodes fail?
